### PR TITLE
Select: fix onOpen called twice

### PR DESCRIPTION
**Pull request by netochaves**
#### What does this PR do?
Fix a bug that `onOpen` is called twice
#### Where should the reviewer start?
`src/js/components/Select/Select.js`
#### What testing has been done on this PR?
Unit
#### How should this be manually tested?
Storybook
#### What are the relevant issues?
#4283 
#### Do the grommet docs need to be updated?
no
#### Should this PR be mentioned in the release notes?
yes
#### Is this change backwards compatible or is it a breaking change?
backwards compatible

### CODE REVIEW

**Comment by netochaves** [comment]
Hey, I changed the `onOpen` and `onClose` tests. before they are a single test, then separate them into two tests because I think that since they are two different props they need to be tested in isolation. What do you guys think?

**Comment by ShimiSun** [comment]
>Hey, I changed the onOpen and onClose tests. before they are a single test, I separate them into two tests because I think that since they are two different props they need to be tested in isolation. What do you guys think?

Agreed.

Adding @JoeButler7 to the discussion.

**Comment by JoeButler7** [comment]
> Hey, I changed the `onOpen` and `onClose` tests. before they are a single test, I separate them into two tests because I think that since they are two different props they need to be tested in isolation. What do you guys think?

I agree, it's always good to separate and isolate things for testing

File: src/js/components/Select/Select.js [hunk 1]
```diff
@@ -264,7 +265,6 @@ const Select = forwardRef(
                   value={inputValue}
                   size={size}
                   theme={theme}
-                  onClick={disabled === true ? undefined : onRequestOpen}
```
Comment (ShimiSun) [inline src/js/components/Select/Select.js:267]: Why was this line removed?

Comment (netochaves) [reply src/js/components/Select/Select.js:267]: Because this event was already being called here:
https://github.com/grommet/grommet/blob/07203885d5da564695390038b09c8ddb9d81044e/src/js/components/Select/Select.js#L206

That's why the `onOpen` was being called twice.

**Comment by ShimiSun** [comment]
Hi @netochaves, Thank you for your contribution! 
I'm getting 'permission denied' error when trying to push the resolved conflicts to your branch.
Can you either grant me permission on your fork or resolve the conflicts so we can merge the PR?
```
➜  GMay git:(4e49ce61f) git push netoc netoc/Select/fix-onOpen-bug
Total 0 (delta 0), reused 0 (delta 0)
To https://github.com/netochaves/grommet
 ! [remote rejected]     netoc/Select/fix-onOpen-bug -> netoc/Select/fix-onOpen-bug (permission denied)
error: failed to push some refs to 'https://github.com/netochaves/grommet'
```

**Comment by netochaves** [comment]
Done @ShimiSun

**Comment by ShimiSun** [comment]
Thank you for your awesome-ness @netochaves !!! 🚀
